src/librustc_resolve/diagnostics.rs

```diff
@@ -725,17 +726,19 @@ impl<'a> Resolver<'a> {
         }
     }
 
-    fn lookup_import_candidates_from_module<FilterFn>(
+    fn lookup_import_candidates_from_module<FilterFn, CollectFn>(
         &mut self,
         lookup_ident: Ident,
         namespace: Namespace,
         parent_scope: &ParentScope<'a>,
         start_module: Module<'a>,
         crate_name: Ident,
         filter_fn: FilterFn,
+        mut collect_fn: CollectFn,
```

Comment:
I think that we could use `impl Trait` here, but it is a stylistic choice that we might or might not want.

```suggestion
        filter_fn: Fn(Res) -> bool,
        mut collect_fn: for<'b> FnMut(&'b Ident, Namespace, &'b NameBinding<'a>),
```

Reply:
It's very helpful. Done.